### DISCUSSION TITLE
It is possible to run .NET Aspire without Docker/Podman? Maybe using Kaniko/Skaffold?

### DISCUSSION THREAD

**Discussion by eliashdezr** [microsoft/aspire#6445] [General]
If for some reason I have restrictions to have docker and podman in my windows machine, is there a way to get around this?

**Comment by davidfowl** [comment]
We only test with docker and podman. If those container runtimes are docker compatible then it might "just work", but we don't officially support them. I don't see how Kaniko/Skaffold are related though. Those aren't container runtimes.

**Reply by eliashdezr** [reply]
@davidfowl  thanks for the response!

Thanks for the clarification! My idea was to build the required containers in advance (using Kaniko) and then point Aspire to use those pre-built images when it runs.

The idea is that since Aspire needs Docker/Podman for building and running containers, I’d just handle the build part separately beforehand. Once the images are built and available (in a local or remote registry), Aspire could just pull them directly when needed.

I know this isn’t the usual setup, but do you think this approach could work?

Regards

**Reply by davidfowl** [reply]
What container runtime are those images going to use if you can't use docker or podman? Are you saying you can use them for running but not building? 

>The idea is that since Aspire needs Docker/Podman for building and running containers, I’d just handle the build part separately beforehand. Once the images are built and available (in a local or remote registry), Aspire could just pull them directly when needed.

Yes you can use AddContainer to use an existing container image.

**Reply by MithunkumarChopda** [reply]
@davidfowl Where is the source code that makes up the aspire dashboard image? Is that https://github.com/dotnet/aspire/tree/main/src/Aspire.Dashboard ?

**Reply by davidfowl** [reply]
Yes
